### PR TITLE
upload assignment 4

### PR DESCRIPTION
**Pull request by ritasamir**

### CODE REVIEW

File: ritasamir/assignment4/CountingIslands.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	 *            is a 2-dimensional array of booleans, where false means water
+	 *            and true means land
+	 * @param rowNumber
+	 *            is the number of rows
+	 * @param columnNumber
+	 *            is the number of columns
+	 * @return the number of islands.
+	 */
+	public int getNumberOfIslands(boolean[][] tiles, int rowNumber,
+			int columnNumber) {
```
Comment (imorlowska) [inline ritasamir/assignment4/CountingIslands.java]: Won't the last parameter fit in the previous line?

File: ritasamir/assignment4/CountingIslands.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+		if (tiles.length != rowNumber || tiles[0].length != columnNumber)
+			throw new Error("the size doesn't match");
+		int counter = 0;
+		this.rowNumber = rowNumber;
+		this.columnNumber = columnNumber;
+		for (int i = 0; i < rowNumber; i++) {
+			for (int j = 0; j < columnNumber; j++) {
+				if (tiles[i][j]) {
+					counter++;
```
Comment (imorlowska) [inline ritasamir/assignment4/CountingIslands.java]: nit: Maybe add a comment explaining why this works. We know the algorithm already, but if someone who had no idea what the solution was were to read this, they might wonder what's going on. :)

File: ritasamir/assignment4/CountingIslands.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+	}
+
+	private void dfs(int row, int col, boolean[][] tiles) {
+		tiles[row][col] = false;
+		Iterator<Point> itr = getPossibleNeighbours(tiles, row, col).iterator();
+		while (itr.hasNext()) {
+			Point point = itr.next();
+			dfs(point.x, point.y, tiles);
+		}
+
```
Comment (imorlowska) [inline ritasamir/assignment4/CountingIslands.java]: nit: remove extra lines

File: ritasamir/assignment4/CountingIslands.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+	 * @param tiles
+	 *            is a 2-dimensional array of booleans, where false means water
+	 *            and true means land
+	 * @param rowNumber
+	 *            is the number of rows
+	 * @param columnNumber
+	 *            is the number of columns
+	 * @return the number of islands.
+	 */
+	public int getNumberOfIslands(boolean[][] tiles, int rowNumber,
```
Comment (imorlowska) [inline ritasamir/assignment4/CountingIslands.java]: You can make this method static and then pass rowNumber and columnNumber as parameters to other methods (or better, just use the size of the boolean array instead).

File: ritasamir/assignment4/CountingIslands.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+	 * @param tiles
+	 *            is a 2-dimensional array of booleans, where false means water
+	 *            and true means land
+	 * @param rowNumber
+	 *            is the number of rows
+	 * @param columnNumber
+	 *            is the number of columns
+	 * @return the number of islands.
+	 */
+	// looping through the 2d array and search for an island when we find a land
```
Comment (imorlowska) [inline ritasamir/assignment4/CountingIslands.java]: nit: Move into inside of the method. Using comments with "//" is okay inside the method, but above the method there should only be the javadoc with the "/** * */" style.

Comment (ritasamir) [reply ritasamir/assignment4/CountingIslands.java]: thank you, i will do it

**Pull request merged**